server/controllers/personal/personalController.ts

```diff
@@ -343,6 +343,85 @@ export default class PersonalController {
     }
   }
 
+  textInput(fieldData: TextFieldData) {
+    const { pageTitle, hintText, auditPage, fieldName, url, inputClasses } = fieldData
+
+    return {
+      edit: async (req: Request, res: Response, next: NextFunction) => {
+        const { prisonerNumber } = req.params
+        const { clientToken, prisonerData } = req.middleware
+        const { firstName, lastName } = prisonerData
+        const requestBodyFlash = requestBodyFromFlash<{ [fieldName: string]: string }>(req)
+        const errors = req.flash('errors')
+        const prisonerBannerName = formatName(firstName, null, lastName, { style: NameFormatStyle.lastCommaFirst })
+        const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
+
+        // TODO remove when API returns value
+        if (prisonPerson) {
+          prisonPerson.physicalAttributes.shoeSize = '7'
+        }
+
+        const fieldValue = requestBodyFlash ? requestBodyFlash[fieldName] : prisonPerson?.physicalAttributes[fieldName]
+
+        await this.auditService.sendPageView({
+          user: res.locals.user,
+          prisonerNumber: prisonerData.prisonerNumber,
+          prisonId: prisonerData.prisonId,
+          correlationId: req.id,
+          page: auditPage,
+        })
+
+        res.render('pages/edit/singleField', {
+          pageTitle: `${pageTitle} - Prisoner personal details`,
+          prisonerNumber,
+          breadcrumbPrisonerName: prisonerBannerName,
+          errors: hasLength(errors) ? errors : [],
```

Comment:
Can just make this return `errors` directly as `req.flash('errors')` will return `[]` if it can't find `errors` in the flash